### PR TITLE
Move from nosetest to pytest

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

The original motivation for this PR was found in the previous PR #2321 - Python code unit test coverage wasn't properly being uploaded to codecov. After a little poking, it seems that it's not being generated properly. Should be easy enough. I was also inspired to look at moving from nose to pytest at the same time.

So this is the first step (of likely many, TBA) in updating the Python test stack to be a little more modern. Nose is an ancient way of running Python tests. It has served us well, but it is long past time to move on. https://nose.readthedocs.io/en/latest/

Pytest is a framework that's used more commonly in the Python world, and at KBase in other projects. It has good support for test fixtures and mocking, and is under active community support. Best of all, it's basically a drop-in replacement! Seriously, the main change to get this running was to add it to `requirements.txt`, and change the call in `scripts/narrative_backend_tests.sh`. This exposed a number of minor warnings that were also changed - notably, any class in the test directory with `Test` as its first token is treated as a test, which got noisy. So the `util.TestConfig` class is now `util.ConfigTests`, for example.

A few other warnings popped up that were trivial to change, as well.

This does alter the `requirements.txt` file, so be sure to do a new installation locally in order to run tests.

- [n/a] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
* REINSTALL THE NARRATIVE! Making this loud and bold here.
* Run tests as usual.
- [ ] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)

- [-] [Version has been bumped](https://semver.org/) for each release
- [x] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: src/biokbase/narrative/common/log_proxy.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def __init__(self, input_file):
             if not isinstance(input_file, IOBase):
                 input_file = open(str(input_file), "r")
             try:
-                self._obj = yaml.load(input_file)
+                self._obj = yaml.load(input_file, Loader=yaml.SafeLoader)
```
Comment (briehl) [inline src/biokbase/narrative/common/log_proxy.py:64]: Another deprecation warning. Really, both `log_proxy` and `kblogging` aren't in use, I don't believe, but this was safe enough.

File: src/requirements.txt [hunk 2]
```diff
@@ -6,7 +6,6 @@ jinja2==2.11.3
 jsonschema==3.2.0
 markupsafe==1.1.1
 mock==4.0.3
-nose>=1.1.2
```
Comment (briehl) [inline src/requirements.txt:9]: Bye bye nose!

File: src/biokbase/narrative/tests/narrative_mock/mockclients.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from ..util import TestConfig
+from ..util import ConfigTests
```
Comment (briehl) [inline src/biokbase/narrative/tests/narrative_mock/mockclients.py:1]: The remaining changes go from `TestConfig` -> `ConfigTests` in the rest of these files.

**Review by ialarmedalien — APPROVED** [review]
lgtm

**Pull request merged**